ete4/smartview/ete/gardening.py

```diff
@@ -0,0 +1,62 @@
+"""
+Tree-related operations.
+
+Sorting, changing the root to a node, moving branches, removing (prunning)...
+"""
+
+# "Arboriculture" may be more precise than "gardening", but it's a mouthful :)
+
+from ete4.smartview.tree import update_size
+from ete4 import Tree
+
+
+def sort(tree, key=None, reverse=False):
+    "Sort the tree in-place"
+    key = key or (lambda node: (node.size[1], node.size[0], node.name))
+    for node in tree.children:
+        sort(node, key, reverse)
+    tree.children.sort(key=key, reverse=reverse)
+
+
+def root_at(node):
+    "Return the tree of which node is part of, rerooted at the given node"
+    root = node.get_tree_root()
+    root.set_outgroup(node)
+    new_root = root.get_tree_root()
+    update_size(new_root)
+    return new_root
```

Comment:
It's probably not important at this point, but I tried from the gui and it didn't work for me (I got a server error related to "Cannot set myself as outgroup").

If that was fixed, the part of `update_size(new_root)` seems suspicious, since it seems we would need to also update the size for all the intermediate nodes that have changed parenthood.

Reply:
In me trying to make use of the `update_size()` function when loading a tree (`tree_load()` in stv's `server.py`) and using the ETE4 Tree class (which does not include `update_size()` in its constructor as opposed to the stv's Tree class) in this new update, I had to make this function recursive, beginning from the root. I do not know if this is the best option since: 
* It is used only in the smartview module both in `server.py` when loading a tree and here in `root_at` (from the top of my head)
* It is modifying an already present property in the Tree class `size`. Maybe we want to include this new behaviour in the whole Tree class as a method? If so, it may make sense to include it also in the constructor as @jordibc did in the stv Tree class

Reply:
I get the server error too!!
I am looking into it because it was previously solved... the tree class method `.get_tree_root()` is not working properly... maybe it has to do with my tampering with the constructor. In progress...

Reply:
I think @jhcepas prefers to have the `size` part separate, but I found it very tied to most operations I wanted to do with the tree (which were mostly drawing), and if I had it separate then I had to do quite a bit of extra gymnastics just to track two values (size[0] and size[1]) all the time.

Seeing that `update_size()` may have changed in meaning (now being a recursive update, as opposed to just updating from the up-to-date size of the children), that part of the code is probably correct. I was assuming it was the old definition :smiley: 

By the way, `update_size()` is not really in the constructor of stv's Tree class, but `size` is indeed a member and `update_size()` is used in the constructor, which is what I think you mean :)